### ISSUE TITLE
Add support for Minecraft version provided Java versions

### ISSUE THREAD

**Issue by RyanTheAllmighty** [ATLauncher/ATLauncher#459]
As of the latest snapshot, Mojang are now providing a `javaVersion` property:

https://launchermeta.mojang.com/v1/packages/256f291742441e215cf6743e4aba6c994b292550/21w11a.json

```
"javaVersion": {
    "component": "jre-legacy",
    "majorVersion": 8
}
```

I'm unsure where to reference this `jre-legacy` but did find https://launchermeta.mojang.com/v1/products/java-runtime/2ec0cc96c44e5a76b9c8b7c39df7210883d12871/all.json

I'm unsure if that url is going to be updated constantly or if there's a `version_manifest.json` equivalent that the vanilla launcher uses to index and get this url.

**Comment by RyanTheAllmighty** [comment]
This needs to go out before 1.17 release otherwise we're going to have a lot of support issues

**Comment by RyanTheAllmighty** [comment]
Released with version 3.4.4.0 of the launcher
